src/Actions/BaseAction.php

```diff
@@ -90,4 +142,24 @@ protected function convertEmptyStringToNull(array $data): array
             return $value === '' ? null : $value;
         }, $data);
     }
+
+    protected function fireActionEvent(string $event, object $object = null, bool $halt = true)
```

Comment:
i think it would be better to always have the Action class in the event and set the return value to a property in the execute method. this gives more context for the event listener